### PR TITLE
doc/user: [Kafka sources] add confluent schema registry basic auth

### PR DESCRIPTION
**Pull request by joacoc**
### Motivation

This PR adds the `Confluent Schema Registry` SASL plain mechanism to the Kafka source guide.

Screen Shot 2022-10-17 at 20 31 45

### Tips for reviewer

It works well with Confluent Schema Registry, but I haven't tested if it works with local deployments of the confluent schema registry.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  -

### CODE REVIEW

**Comment by morsapaes** [comment]
What makes this `SASL` versus `HTTP Basic`? Are we inheriting the `SASL` auth for Schema Registry if the broker connection is configured to use `SASL`? Asking because we refer to just using `username` and `password` as HTTP basic authentication in the [`CREATE CONNECTION` page](https://materialize.com/docs/sql/create-connection/#confluent-schema-registry).

**Comment by joacoc** [comment]
@morsapaes `SASL` is, in fact, wrong. After checking the CSSR client, I can say it makes a basic auth request:

https://github.com/MaterializeInc/materialize/blob/f4ed657b7034660c28ef82083214405d7d929bca/src/ccsr/src/client.rs#L54

File: doc/user/content/sql/create-source/kafka.md [hunk 1]
```diff
@@ -320,6 +320,18 @@ CREATE CONNECTION csr_ssl TO CONFLUENT SCHEMA REGISTRY (
 );
 ```
 {{< /tab >}}
+{{< tab "Basic Authentication">}}
```
Comment (morsapaes) [inline doc/user/content/sql/create-source/kafka.md:323]:
```suggestion
{{< tab "Basic HTTP Authentication">}}
```

Comment (morsapaes) [reply doc/user/content/sql/create-source/kafka.md:323]: For [standardization](https://www.rfc-editor.org/rfc/rfc7617).

File: doc/user/content/sql/create-source/kafka.md [hunk 2]
```diff
@@ -320,6 +320,18 @@ CREATE CONNECTION csr_ssl TO CONFLUENT SCHEMA REGISTRY (
...
+```sql
+CREATE SECRET IF NOT EXISTS csr_username AS '<CSR_USERNAME>';
+CREATE SECRET IF NOT EXISTS csr_password AS '<CSR_PASSWORD>';
+
+CREATE CONNECTION csr_sasl
```
Comment (morsapaes) [inline doc/user/content/sql/create-source/kafka.md:328]:
```suggestion
CREATE CONNECTION csr_basic_http
```

**Review by morsapaes — CHANGES_REQUESTED** [review]

**Review by morsapaes — APPROVED** [review]

**Pull request merged**